### ISSUE TITLE
Proposal: force upgrade to same k8s version

### ISSUE THREAD

**Issue by serbrech** [Azure/aks-engine#522]
**Is this a request for help?**:  
No

---
**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Feature Proposal

---

**description**:
We want to be able to force the upgrade process on a cluster without changing the kubernetes version. This will ensure that all components of the cluster are healthy and using the latest validated version for a given kubernetes version.

**proposed cli**: 
For an existing kubernetes `1.12.5` cluster, we want the following command to go through each nodes of the cluster and "refresh" them if necessary (update to latest valid version).

```bash
> aks-engine upgrade --force --location westus --resource-group rg --subsctiption $sub --upgrade-version 1.12.5
```

**Open questions**: 
- What should the behavior of the `--force` flag be for a usual next k8s version upgrade?
- Should the --force flag allow other arbitrary upgrade (not valid per our `get-versions` matrix)? 

`--force` makes it seem like you can _force_ the upgrade no matter what. However, a forbidden upgrade path (not in the `get-versions` mattrix) should still be forbidden I suppose?

**Related**
https://github.com/Azure/acs-engine/pull/3810

**Comment by jackfrancis** [comment]
I suppose the most graceful way to do this is to support "downgrade" as well. In other words, any version that is supported for create can be the target version, including versions "lower" than the current version on the cluster.

I think that would in fact offer more help to folks, in scenarios where an upgrade brought undesired breakage.

**Comment by mboersma** [comment]
I think ideally `--force` would allow you to bypass our normal upgrade rules and target any supported version (allowing "downgrade" as well). We should print some warning in that case, however, because the chances of a downgrade breaking deployed apps are much higher.

**Comment by serbrech** [comment]
@mboersma this is what we are going for

**Comment by CecileRobertMichon** [comment]
fixed in #525
